File: other/Solution9.java
public class Solution9 {
    public boolean isPalindrome(int x) {
        if(x < 0){
            return false;
        }
        int copy = x;
        int reverse = 0;
        while(copy != 0){
            reverse  = reverse * 10;
            reverse += copy % 10;
            copy = copy / 10;
        }
        return x == reverse;
    }
}

File: C++/Solution9.cpp
#include<iostream>
#include<vector>

using namespace std;

class Solution9 {
public:
    bool isPalindrome(int x) {
        if(x < 0){
            return false;
        }
        return x == reverse(x);
    }

    int reverse(int x){
        long result = 0;
        while(x > 0){
            result = 10 * result + x % 10;
            x = x / 10;
        }
        if(result > INT_MAX){
            return -1;
        }
        return result;
    }
};

int main(){
    Solution9 s;
    int x = INT_MAX;
    cout << s.isPalindrome(x);
    return 0;
}